### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by Mr-Blue-Sky-Candy** [InsightSoftwareConsortium/ITK#4604]
I only get segmentation fault. I can share the code I wrote but not the Dicom series. The code I wrote works for all others but not for some particular CT scans. How can I help you resolve this?

The code I wrote is the following:
```
names_generator = itk.GDCMSeriesFileNames.New()
names_generator.SetUseSeriesDetails(True)
names_generator.SetGlobalWarningDisplay(True)
names_generator.SetNumberOfWorkUnits(os.cpu_count())
names_generator.SetLoadPrivateTags(False)
names_generator.DebugOn()

dicomIO = itk.GDCMImageIO.New()
dicomIO.LoadPrivateTagsOff()
dicomIO.SetGlobalWarningDisplay(True)
reader = itk.ImageSeriesReader[itk.Image[itk.ctype("int"), 3]].New()
reader.SetGlobalWarningDisplay(True)
reader.SetMetaDataDictionaryArrayUpdate(True)
reader.SetNumberOfWorkUnits(os.cpu_count())
reader.SetImageIO(dicomIO)
reader.ForceOrthogonalDirectionOff()
```

It gives segmentation fault at the line
`names_generator.SetDirectory(folder)`

This code also gives segmentation fault:
```
reader = sitk.ImageSeriesReader()
dicom_names = reader.GetGDCMSeriesFileNames(folder)
reader.SetFileNames(dicom_names)
image = reader.Execute()
```

Pydicom is able to read it but its metadata is very messy and hard to format besides it cannot read other particular Dicom series of CT scans.

ITK version is 5.3 on both ubuntu 22 and archlinux

**Comment by Mr-Blue-Sky-Candy** [comment]
I agree it is due to binded cpp or c code but I cannot share the dicom files. I know how to code cpp as well but I don't know how to debug a python binding. If there is a way I can help about this, I would be happy to do so

**Comment by dzenanz** [comment]
Can you try C++ code from [this example](https://examples.itk.org/src/io/gdcm/readdicomseriesandwrite3dimage/documentation) on your DICOM data? Does it also crash?

**Comment by Mr-Blue-Sky-Candy** [comment]
> Hi, could you please try with `itk-5.4rc4`? Do the same results occur?

this also failed

**Comment by Mr-Blue-Sky-Candy** [comment]
> Also,
> 
> ```python
> image = itk.imread('path/to/dicom/dir/')
> ```

I already tried but forgot to mention. The outcome is the same

**Comment by thewtex** [comment]
If you set up a debug build and get a backtrace, it will likely be evident where the issue exists. Please see [Section 9.5.3 in The ITK Software Guide, Book 1](https://github.com/InsightSoftwareConsortium/ITKSoftwareGuide/releases/download/v5.3.0/InsightSoftwareGuide-Book1-5.3.0.pdf) for guidance.

**Comment by Mr-Blue-Sky-Candy** [comment]
Hey, I had to refresh my cpp skills. I ran with debug mode but I always used qt creator run debug mode. So I can see this:
                                                                                                                     
```                                                                                                                     
1  ??                                                                                                 0x7ffff4cab32c 
2  raise                                                                                              0x7ffff4c5a6c8 
3  abort                                                                                              0x7ffff4c424b8 
4  std::__glibcxx_assert_fail                                             debug.cc                 61 0x7ffff4edd3b2 
5  ??                                                                                                 0x7ffff654881e 
6  gdcm::PixmapReader::ReadImageInternal(gdcm::MediaStorage const&, bool)                             0x7ffff654ab16 
7  gdcm::ImageReader::ReadImage(gdcm::MediaStorage const&)                                            0x7ffff654c880 
8  gdcm::PixmapReader::Read()                                                                         0x7ffff65472da 
9  gdcm::SerieHelper::AddFileName(std::string const&)                                                 0x7ffff6522c7d 
10 gdcm::SerieHelper::SetDirectory(std::string const&, bool)                                          0x7ffff6522f3c 
11 itk::GDCMSeriesFileNames::SetInputDirectory(std::string const&)                                    0x7ffff698ebb4 
12 itk::GDCMSeriesFileNames::SetDirectory                                 itkGDCMSeriesFileNames.h 98 0x555555567b5d 
13 main                                                                   main.cpp                 32 0x555555566dc5 
```
Is this good?
It also show disassembler output. If you need me to share something else, let me know. I don't know how to proceed from this point on:
![2023-04-22_21-57](https://github.com/InsightSoftwareConsortium/ITK/assets/131680576/ce26fc9c-13e1-45a6-b853-87144b90852d)

**Comment by thewtex** [comment]
Nice!

It would be helpful to build as `Debug` `CMAKE_BUILD_TYPE` and find out the line number that identifies which `assert` is failing:

https://github.com/InsightSoftwareConsortium/ITK/blob/596ed5fd6dd52f0beb2c6af6147015fe09923736/Modules/ThirdParty/GDCM/src/gdcm/Source/MediaStorageAndFileFormat/gdcmPixmapReader.cxx#L623-L1132

CC @malaterre

**Comment by Mr-Blue-Sky-Candy** [comment]
I can see this in the command line output:
```
$ ./test_itk ../../../../work/AIRTQA/r-008-023-2-corrupt/
/usr/include/c++/13.2.1/bits/stl_vector.h:1125: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](size_type) [with _Tp = gdcm::Curve; _Alloc = std::allocator<gdcm::Curve>; reference = gdcm::Curve&; size_type = long unsigned int]: Assertion '__n < this->size()' failed.
Aborted (core dumped)
```

**Comment by Mr-Blue-Sky-Candy** [comment]
I am already building Debug builds but as it is described in ITK software guide I switched to RelWithDebInfo build anyway. Unsurprisingly, the result is the same.

**Comment by thewtex** [comment]
@Mr-Blue-Sky-Candy there are no line numbers for the screenshot shared in the GDCM calls -- is `ITK_USE_SYSTEM_GDCM` `OFF`?

**Comment by thewtex** [comment]
The setting should be:

> set(ITK_USE_SYSTEM_GDCM OFF)

So GDCM with a Debug build is used. Examining with `gdb` may be helpful as suggested in the ITK Software Guide.

**Comment by Mr-Blue-Sky-Candy** [comment]
I am already running in qtcreator with gdb.

> @Mr-Blue-Sky-Candy there are no line numbers for the screenshot shared in the GDCM calls -- is `ITK_USE_SYSTEM_GDCM` `OFF`?

Actually there is. I will share a different view with `set(ITK_USE_SYSTEM_GDCM OFF)`:
![image](https://github.com/InsightSoftwareConsortium/ITK/assets/131680576/e868d8db-bc1f-44b4-8687-c68449441d85)

**Comment by thewtex** [comment]
We are looking for what is happening at Level 7:

![image](https://github.com/InsightSoftwareConsortium/ITK/assets/25432/633c4117-faf8-4e35-8a4b-adb5ae14a634)

**Comment by Mr-Blue-Sky-Candy** [comment]
Ahh OK. Thanks. I recompiled gdcm with debug symbols. Here you go:
![image](https://github.com/InsightSoftwareConsortium/ITK/assets/131680576/4b87cd14-1a9e-46dc-b3e4-1230ad054b03)

**Comment by Mr-Blue-Sky-Candy** [comment]
All functions look better here
![image](https://github.com/InsightSoftwareConsortium/ITK/assets/131680576/f369fb8f-167c-4856-a0bc-a94574345d75)

**Comment by thewtex** [comment]
Thanks, closer!

You will need to step through the code in gdcmPixmapReader (set a breakpoint in QtCreater) to understand what content that is expected is not present.

**Comment by Mr-Blue-Sky-Candy** [comment]
![image](https://github.com/InsightSoftwareConsortium/ITK/assets/131680576/3ee6e571-c36d-438e-938b-0d1de59b7129)
Here `return Curves[i]`, i is 7 but it is larger than size of `Curves`

**Comment by thewtex** [comment]
@malaterre any suggestions?

**Comment by hjmjohnson** [comment]
@Mr-Blue-Sky-Candy   We have reached the end of support suggestions on this issue without having the ability to recreate the problem.  Please re-open this issue if more information can be provided.
